src/_reducers/departments.reducer.js

```diff
@@ -48,7 +51,8 @@ export default (state = initialState.departments, action) => {
         ...state,
         departmentsList: action.payload,
         departmentsCount: action.payload.length,
-        isLoading: false
+        isLoading: false,
+        departmentDetail: {}
```

Comment:
This is not good. You should have seperate reducers - one for all departments, another for department details and use combine reducers to return a single reducer for departments.